ngx_http_upstream_check_module.c

```diff
@@ -241,6 +242,7 @@ struct ngx_http_upstream_check_srv_conf_s {
     ngx_array_t                             *fastcgi_params;
 
     ngx_uint_t                               default_down;
+    ngx_uint_t                               ssl;
```

Comment:
Yes, it's a boolean. Unfortunately, Nginx doesn't define (or use the standard) boolean type. I think using bit fields would make sense for a collection of flags, not for a single one.